src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtl.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Raven.Client.Documents.Commands.Batches;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Http;
+using Raven.Server.Documents.ETL.Providers.Raven.Enumerators;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.ETL.Providers.Raven
+{
+    public class RavenEtl : EtlProcess<RavenEtlItem, ICommandData>
+    {
+        public const string RavenEtlTag = "Raven ETL";
+
+        public RavenEtl(DocumentDatabase database, RavenEtlConfiguration configuration) : base(database, configuration, RavenEtlTag)
+        {
+            EtlConfiguration = configuration;
+        }
+
+        public RavenEtlConfiguration EtlConfiguration { get; }
+
+        protected override IEnumerator<RavenEtlItem> ConvertDocsEnumerator(IEnumerator<Document> docs)
+        {
+            return new DocumentsToRavenEtlItems(docs);
+        }
+
+        protected override IEnumerator<RavenEtlItem> ConvertTombstonesEnumerator(IEnumerator<DocumentTombstone> tombstones)
+        {
+            return new TombstonesToRavenEtlItems(tombstones);
+        }
+
+        protected override EtlTransformer<RavenEtlItem, ICommandData> GetTransformer(DocumentsOperationContext context)
+        {
+            return new RavenEtlDocumentTransformer(Database, context, EtlConfiguration);
+        }
+
+        protected override void LoadInternal(IEnumerable<ICommandData> commands, JsonOperationContext context)
+        {
+            using (var requestExecutor = RequestExecutor.ShortTermSingleUse(EtlConfiguration.Url, EtlConfiguration.Database, EtlConfiguration.ApiKey)) // TODO arek - consider caching it somewhere
```

Comment:
We want to use a long term request executor here, because we'll be calling this a lot, and we want to save the auth and connection tearing.